app/lib/efile/nj/nj_retirement_income_helper.rb

```diff
@@ -10,6 +10,38 @@ def initialize(intake)
         end
       end
 
+      def show_retirement_income_warning?(line_14, line_15, line_16a)
+        if non_military_1099r_box_1_total.zero?
+          return false
+        end
+
+        box_1_gross_income = non_military_1099r_box_1_total + line_15 + line_16a
+        max_exclusion_threshold = if @intake.filing_status_mfs? 
+                                    50_000
+                                  elsif @intake.filing_status_mfj?
+                                    100_000
+                                  else
+                                    75_000
+                                  end
+        if box_1_gross_income <= max_exclusion_threshold
+          return false
+        end 
+
+        if non_military_1099r_box_1_total <= max_exclusion_threshold && (line_14 > 3_000 || all_filers_under_62?)
+          return false
+        end
+
+        true
+      end
+
+      def non_military_1099r_box_1_total
+        box_1_totals = 0
+        @non_military_1099rs.each do |non_military_1099r|
+          box_1_totals += non_military_1099r.gross_distribution_amount.round
+        end
+        box_1_totals
+      end
```

Comment:
[pebble] this can be simplified to

```
def non_military_1099r_box_1_total
  @non_military_1099rs.sum { |non_military_1099r| non_military_1099r.gross_distribution_amount.round }
end
```